ui/app/controllers/clients/client/index.js

```diff
@@ -277,4 +281,59 @@ export default class ClientController extends Controller.extend(
       this.set('activeTask', null);
     }
   }
+
+  // #region metadata
+
+  @tracked editingMetadata = false;
+
+  get hasMeta() {
+    return (
+      this.model.meta?.structured && Object.keys(this.model.meta?.structured)
+    );
+  }
+
+  @tracked newMetaData = {
+    key: '',
+    value: '',
+  };
+
+  @action resetNewMetaData() {
+    this.newMetaData = {
+      key: '',
+      value: '',
+    };
+  }
+
+  @action validateMetadata(event) {
+    if (event.key === 'Escape') {
+      this.resetNewMetaData();
+      this.editingMetadata = false;
+    }
+  }
+
+  @action async addDynamicMetaData({ key, value }, e) {
+    try {
+      e.preventDefault();
+      await this.model.addMeta({ [key]: value });
+
+      this.flashMessages.add({
+        title: 'Metadata added',
+        message: `${key} successfully saved`,
+        type: 'success',
+        destroyOnClick: false,
+        timeout: 3000,
+      });
+    } catch (err) {
+      const error =
+        messageFromAdapterError(err) || 'Could not save new dynamic metadata';
```

Comment:
`nit`: `messageFromAdapterError` never returns `void` because it returns `Unknown error` if no cases are met. You can remove the logical OR operator.